sdk/tables/azure-data-tables/src/table_clients.cpp

```diff
@@ -511,14 +510,11 @@ Azure::Response<Models::MergeEntityResult> TableClient::MergeEntity(
   request.SetHeader(ContentLengthHeader, std::to_string(requestBody.Length()));
   request.SetHeader(AcceptHeader, AcceptFullMeta);
   request.SetHeader(PreferHeader, PreferNoContent);
-  if (!tableEntity.GetETag().Value.empty())
+
+  if (isUpsert == false && !tableEntity.GetETag().Value.empty())
```

Comment:
This is a complete nit, but I generally prefer `if (!isUpsert...` but this is 100% up to you.